cmake will only run properly named c10 tests

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98669
* #98712
* #98711
* #98651
* #98931
* #98987
* #98949
* #98930
* #98953
* #98598
* #98597
* #98650
* #98649
* #98648
* __->__ #98710

For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

Commit:
cmake will only run properly named c10 tests

For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

PaliC reviewed (approved):
lgtm, all tests seem to follow this rule as well already.

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]

Commit:
Update on "cmake will only run properly named c10 tests"


For the purpose of our Bazel and Meta-internal macros tests, we want
to create a single binary that can verify the different
configurations. CMake would see this file and try to run it and fail
in Windows which uses different values.

But we don't care about verifying this in CMake since it's not part of
the build unification effort.

In order to do this, we have to rename the SmallVectorTest to match
the naming convention of the rest of the c10 tests.

Differential Revision: [D44823440](https://our.internmc.facebook.com/intern/diff/D44823440/)

[ghstack-poisoned]
